### PR TITLE
52 create footer

### LINKED ISSUE #52: create Footer

### PR DESCRIPTION
**Pull request by sergiobtos**
Footer Component created. Also has some lint changes to be added to the code. I used gradient instead image

image

image

### CODE REVIEW

File: src/components/Footer/styles.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  font-size: 0.75rem;
+  font-weight: 300;
+
+  //media query mixins
+  ${m.tablet(css`
+    text-align: end;
+  `)}
+`
+
+export default {
```
Comment (rmc3408) [inline src/components/Footer/styles.js]: please, use regular convention
`export const Footer`
`export const Container`

so no need to export default.
export default is used to export a single value as the default export, while export with named exports is used to export multiple values.

File: src/components/Footer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Logo } from '@/src/assets'
+import S from './styles'
```
Comment (rmc3408) [inline src/components/Footer/index.tsx]: See the comment in the next file about export.

`import { A, B, C }  from styles`

File: src/assets/index.tsx [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default as Logo } from './logo'
```
Comment (rmc3408) [inline src/assets/index.tsx]: Could put `assest/svg/index.tsx` folder. and we will put all SVG inside there.

File: src/components/Footer/styles.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable import/no-anonymous-default-export */
```
Comment (rmc3408) [inline src/components/Footer/styles.js]: need to be TS file. But check if need to be TSX.

File: src/components/Footer/styles.js (hunk 1 above)
Comment (rmc3408) [reply src/components/Footer/styles.js]: and remove eslint, since you eslint catch a anonymous default.

File: src/components/Footer/styles.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import { css } from '@emotion/react'
+import styled from '@emotion/styled'
+import * as m from '../../styles/_mixins'
+import { $lgDesktopBreakpoint } from '../../styles/_variables'
+
+const Footer = styled.footer`
```
Comment (rmc3408) [inline src/components/Footer/styles.js]: Footer will give you conflitcs to `Footer` from NEXTJs.... just change name for something else... i.e. FooterBackground or FooterSection

File: src/components/Footer/styles.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  background: rgb(240, 8, 235);
+  background: -moz-linear-gradient(
```
Comment (rmc3408) [inline src/components/Footer/styles.js]: Can we create in global CSS variable and use here. for example
```
:root {
  --accent-pink: 223;
  --accent-blue: 165;
}

h1 {
  background: linear-gradient(163deg, rgba(255, 255, 255, 0), rgba(var(--accent-pink), var(--accent-blue), 1));
}
```

File: src/components/index.tsx [hunk 7]
```diff
@@ -0,0 +1 @@
+export { default as Footer } from './Footer'
```
Comment (rmc3408) [inline src/components/index.tsx:1]: Good way to import components centralized to index.

again, maybe we discuss about modules (several components), components, and UIstyled folder... to not make a mess.

File: src/styles/_mixins.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import { css } from '@emotion/react'
```
Comment (rmc3408) [inline src/styles/_mixins.js:1]: We need to rewrite this file to TS, change this names. They were used from SCSS. ??? but i like this idea a lot..

Also, since we dont have any responsiveness design... keep only lgDesktop. 
In future, for us "trainning" 3 sizes are good ( mobile, tablet, desktop).

File: src/styles/_variables.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+export const // Media Query Breakpoints
```
Comment (rmc3408) [inline src/styles/_variables.js:1]: rewrite this.

File: src/styles/styles.ts [hunk 10]
```diff
@@ -1,14 +1,15 @@
 import { css, keyframes } from '@emotion/react'
```
Comment (rmc3408) [inline src/styles/styles.ts]: this file can be removed... since remove <Animated /> from PAGE.tsx.

File: src/components/Footer/styles.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    rgba(11, 22, 102, 1) 79%
+  );
+  background: linear-gradient(163deg, rgba(240, 8, 235, 1) 0%, rgba(29, 83, 253, 1) 35%, rgba(11, 22, 102, 1) 79%);
+  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#f008eb",endColorstr="#0b1666",GradientType=1);
+  min-height: 22rem;
+
+  display: flex;
+  flex-wrap: wrap;
+  align-content: flex-end;
+  padding-bottom: 64px;
```
Comment (rmc3408) [inline src/components/Footer/styles.js]: you might need to add padding-right and padding-left here... and make the logo+ cpyright be at end. Now, on my computer look like this.
image

Comment (rmc3408) [reply src/components/Footer/styles.js]: but add TODO or BUG comments for later. this is really minor points.

**Review by rmc3408 — APPROVED** [review]
LGTM. 

**To fix**: Just need some changes to keep same style as everyone. Folder structure need to implement. and those JS files mixin and variable must be rewriteen. Which doesnt need to be now. It is a good practice to split in small tickets and give to someone.

We can bring this after meeting you and I only.

I never used SCSS or whatever is those `$mobile, $args`. Please, let me know.

"**To be considered:**" But we need to be carefull with Post Processing because this is NEXT JS which will create a page in Client or SERVER... can cause lots of hydration or flicking screen.

**In Summary,** I am very happy for you join us and seeing this code. You gonna help me too.

File: src/app/global.css [hunk 12]
```diff
@@ -0,0 +1,20 @@
+* {
+  margin: 0;
```
Comment (rmc3408) [inline src/app/global.css:2]: In MUI, it has cssbaseline or GlobalStyles (which i prefer)

File: src/components/Footer/index.tsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+const Footer = () => {
```
Comment (rmc3408) [inline src/components/Footer/index.tsx:5]: It will get config with pre-built component from NEXT.js... change name.

File: src/components/Footer/styles.js (hunk 1 above)
Comment (sergiobtos) [reply src/components/Footer/styles.js]: Done

File: src/components/Footer/index.tsx (hunk 2 above)
Comment (sergiobtos) [reply src/components/Footer/index.tsx]: done

File: src/assets/index.tsx (hunk 3 above)
Comment (sergiobtos) [reply src/assets/index.tsx]: done

File: src/styles/_mixins.js (hunk 8 above)
Comment (sergiobtos) [reply src/styles/_mixins.js:1]: this is how to use : 
import * as m from '@/styles/_mixins';
${m.transition('all 0.3s ease')};

File: src/styles/_variables.js (hunk 9 above)
Comment (sergiobtos) [reply src/styles/_variables.js:1]: we need to decide on the responsiveness break point. I am going to delete the colors

File: src/components/Footer/styles.js (hunk 11 above)
Comment (sergiobtos) [reply src/components/Footer/styles.js]: TODO comment created.

File: src/styles/_variables.js (hunk 9 above)
Comment (rmc3408) [reply src/styles/_variables.js:1]: I would say keep only 3 Break points, 
mobileBreakpoint
tabletBreakpoint
lgDesktopBreakpoint

File: src/styles/_mixins.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+  }
+`
+
+export const desktop = (content) => css`
+  @media (min-width: ${$desktopBreakpoint}) {
+    ${content}
+  }
+`
+
+export const desktopBreakpointPlus = (content) => css`
```
Comment (rmc3408) [inline src/styles/_mixins.js]: clean this code based on 3 break points  which names are in the other comment box.

File: src/styles/_variables.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  $smMobileBreakpoint = '350px',
+  $mobileBreakpoint = '578px',
+  $lgMobileBreakpoint = '767px',
+  $tabletBreakpoint = '768px',
+  $desktopBreakpointMinus = '1023px',
```
Comment (rmc3408) [inline src/styles/_variables.js]: delete unsed ones

**Review by rmc3408 — APPROVED** [review]
just fix other conversation points.

**Review by sergiobtos — COMMENTED** [review]
good now?
